Tests/ClientTests/Wallpaper/WallpaperDataServiceTests.swift

```diff
@@ -0,0 +1,153 @@
+        } catch {
+            XCTFail("We should not fail the extraction process, but did with error: \(error)")
+        }
+    }
+
+    func testSuccessfullyExtractWallpaperMetadata_WithNoLocales() async {
+        let data = getDataFromJSONFile(named: .noLocales)
+        let expectedMetadata = getExpectedMetadata(for: .noLocales)
+
+        networking.result = .success(data)
+        let sut = WallpaperDataService(with: networking)
+
+        do {
+            let actualMetadata = try await sut.getMetadata()
+            XCTAssertEqual(
+                actualMetadata,
+                expectedMetadata,
+                "The metadata that was decoded from data was not what was expected.")
+        } catch {
+            XCTFail("We should not fail the extraction process, but did with error: \(error)")
+        }
+    }
+
+    func testSuccessfullyExtractWallpaperMetadata_WithOnlyStartingAvailability() async {
+        let data = getDataFromJSONFile(named: .availabilityStart)
+        let expectedMetadata = getExpectedMetadata(for: .availabilityStart)
+
+        networking.result = .success(data)
+        let sut = WallpaperDataService(with: networking)
+
+        do {
+            let actualMetadata = try await sut.getMetadata()
+            XCTAssertEqual(
+                actualMetadata,
+                expectedMetadata,
+                "The metadata that was decoded from data was not what was expected.")
+        } catch {
+            XCTFail("We should not fail the extraction process, but did with error: \(error)")
+        }
+    }
+
+    func testSuccessfullyExtractWallpaperMetadata_WithOnlyEndingAvailability() async {
+        let data = getDataFromJSONFile(named: .availabilityEnd)
+        let expectedMetadata = getExpectedMetadata(for: .availabilityEnd)
+
+        networking.result = .success(data)
+        let sut = WallpaperDataService(with: networking)
+
+        do {
+            let actualMetadata = try await sut.getMetadata()
+            XCTAssertEqual(
+                actualMetadata,
+                expectedMetadata,
+                "The metadata that was decoded from data was not what was expected.")
+        } catch {
+            XCTFail("We should not fail the extraction process, but did with error: \(error)")
+        }
+    }
+
+    func testFailToExtractWallpaperMetadata_WithBadLastUpdatedDate() async {
+        let data = getDataFromJSONFile(named: .badLastUpdatedDate)
+
+        networking.result = .success(data)
+        let sut = WallpaperDataService(with: networking)
+
+        do {
+            _ = try await sut.getMetadata()
+            XCTFail("We should fail the extraction process")
+
+        } catch let error {
+            XCTAssertEqual(error.localizedDescription,
+                           "The data couldn’t be read because it isn’t in the correct format.",
+                           "Unexpected decoding failure")
```

Comment:
Wouldn't it be an expected failure here?
```suggestion
                           "Expected decoding failure when decoding bad last updated date")
```

Reply:
No. I'm expecting an error, but, if the error is not the same as the expected one, then it would be unexpected.